java/pom.xml

```diff
@@ -58,9 +58,77 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-shade-plugin</artifactId>
-        <version>2.3</version>
+        <version>3.1.1</version>
         <configuration>
-          <createDependencyReducedPom>false</createDependencyReducedPom>
+          
+          <!-- The following settings can be used to create a JAR which is -->
```

Comment:
This is a big comment -- how can you make this more concise, and only focus on the things the user _needs_ to know? For example "this setup .. may or may not work" -- is that necessary for all users?

Reply:
Okay! I think I have an idea on how to make it more concise!